util/webhook/certs/certmanager.go

```diff
@@ -0,0 +1,275 @@
+	}
+
+	// Update webhook configurations with new cert data.
+	if err := cm.updateWebhookConfigs(ctx, secretData[CACertName]); err != nil {
+		return time.Time{}, err
+	}
+
+	cm.opts.Logger.Info("Rotated certificates", "nextRotation", nextRotationTime.String(), "totalRotationCount", rotationCount)
+	return nextRotationTime, nil
+}
+
+// getNextRotationTime fetches the next rotation time from the webhook secret's annotation.
+func (cm *CertificateManager) getNextRotationTime(secret *corev1.Secret) (time.Time, error) {
+	// Get the next-rotation annotation
+	szValue := secret.Annotations[cm.opts.NextRotationAnnotationKey]
+	if szValue == "" {
+		return time.Time{}, fmt.Errorf("missing next-rotation annotation")
+	}
+
+	// The error is ignored since the value will always be >0.
+	iValue, _ := strconv.ParseInt(szValue, 10, 64)
+	if iValue <= 0 {
+		return time.Time{}, fmt.Errorf("next-rotation annotation value is invalid")
+	}
+
+	// Return the next-rotation value.
+	return time.Unix(iValue, 0), nil
+}
+
+// updateWebhookConfigs updates the caBundle in the webhooks selected by the label selector.
+func (cm *CertificateManager) updateWebhookConfigs(ctx context.Context, caCertData []byte) error {
+	cm.opts.Logger.Info("Updating webhook configs with CA bundle")
+
+	// Update validating webhooks with new CA data.
+	labelSelector, err := metav1.ParseToLabelSelector(cm.opts.WebhookConfigLabel)
+	if err != nil {
+		return fmt.Errorf("failed to parse webhook label: %w", err)
+	}
+	matchLabels := client.MatchingLabels(labelSelector.MatchLabels)
+	validatingWebhookList := &adminregv1.ValidatingWebhookConfigurationList{}
+	if err := cm.opts.Client.List(ctx, validatingWebhookList, matchLabels); err != nil && !apierrors.IsNotFound(err) {
+		return fmt.Errorf("failed to list validating webhooks: %w", err)
+	}
+
+	for _, webhookConfig := range validatingWebhookList.Items {
+		for i := range webhookConfig.Webhooks {
+			webhookConfig.Webhooks[i].ClientConfig.CABundle = caCertData
+		}
+
+		if err := cm.opts.Client.Update(ctx, &webhookConfig); err != nil {
+			if !apierrors.IsConflict(err) {
+				return fmt.Errorf("failed to update validating webhook configuration %s/%s: %w", webhookConfig.Namespace, webhookConfig.Name, err)
+			}
+		}
+	}
+
+	// Update mutating webhooks with new CA data.
+	mutatingWebhookList := &adminregv1.MutatingWebhookConfigurationList{}
+	if err := cm.opts.Client.List(ctx, mutatingWebhookList, matchLabels); err != nil && !apierrors.IsNotFound(err) {
+		return fmt.Errorf("failed to list mutating webhooks: %w", err)
+	}
+
+	for _, webhookConfig := range mutatingWebhookList.Items {
+		for i := range webhookConfig.Webhooks {
+			webhookConfig.Webhooks[i].ClientConfig.CABundle = caCertData
+		}
+
+		if err := cm.opts.Client.Update(ctx, &webhookConfig); err != nil {
+			if !apierrors.IsConflict(err) {
+				return fmt.Errorf("failed to update mutating webhook configuration %s/%s: %w", webhookConfig.Namespace, webhookConfig.Name, err)
```

Comment:
Same as previous comment: Why are we not continuing with other remaining mutating webhooks even though the current one fails?